pytest_timeout.py

```diff
@@ -158,9 +157,11 @@ def pytest_runtest_protocol(item):
     is_timeout = settings.timeout is not None and settings.timeout > 0
     if is_timeout and settings.func_only is False:
         hooks.pytest_timeout_set_timer(item=item, settings=settings)
-    yield
-    if is_timeout and settings.func_only is False:
-        hooks.pytest_timeout_cancel_timer(item=item)
+    try:
+        yield
+    finally:
+        if is_timeout and settings.func_only is False:
+            hooks.pytest_timeout_cancel_timer(item=item)
```

Comment:
this seems like quite an important bugfix.  could you update the changelog as well?

Reply:
With old style hook wrappers it's incorrect to add try/finally

Reply:
Oh?!

Does the code after the `yield` always run then?

Got a link to something I could read? I have several plugins which might need to be updated...

What's the consequence?

Reply:
For legacy hook wrappers one needs to check the exception attribute if one needs to handle them

Moder hookwrappers Work like expected